comp/core/secrets/secretsimpl/walker.go

```diff
@@ -7,82 +7,51 @@ package secretsimpl
 
 import (
 	"fmt"
-
-	"github.com/DataDog/datadog-agent/comp/core/secrets"
+	"strconv"
 )
 
-type resolverCallback func([]string, string) (string, error)
+type resolveFunc func([]string, string) (string, error)
```

Comment:
Do we still need this type?

Reply:
Nope, good catch, it can be removed.

Reply:
I mean that we might not need a type definition if is not public.

```go
type walker struct {
	// resolver is called to fetch the value of a handle
	resolver func([]string, string) (string, error)
}
```

Reply:
Ah whoops, yes of course.